pytorch_lightning/loggers/neptune.py

```diff
@@ -11,29 +12,80 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
 """
 Neptune Logger
 --------------
 """
+__all__ = [
+    "NeptuneLogger",
+]
+
 import logging
+import operator
+import os
 from argparse import Namespace
-from typing import Any, Dict, Iterable, Optional, Union
+from functools import reduce
+from typing import Any, Dict, Optional, Set, Union
+from weakref import ReferenceType
 
 import torch
-from torch import is_tensor
 
+from pytorch_lightning import __version__
 from pytorch_lightning.loggers.base import LightningLoggerBase, rank_zero_experiment
 from pytorch_lightning.utilities import _module_available, rank_zero_only
+from pytorch_lightning.utilities.imports import _compare_version
+from pytorch_lightning.utilities.model_summary import ModelSummary
+from pytorch_lightning.callbacks.model_checkpoint import ModelCheckpoint
 
-log = logging.getLogger(__name__)
 _NEPTUNE_AVAILABLE = _module_available("neptune")
-
-if _NEPTUNE_AVAILABLE:
-    import neptune
-    from neptune.experiments import Experiment
+_NEPTUNE_GREATER_EQUAL_0_9 = _NEPTUNE_AVAILABLE and _compare_version("neptune", operator.ge, "0.9.0")
+
+if _NEPTUNE_AVAILABLE and _NEPTUNE_GREATER_EQUAL_0_9:
+    try:
+        from neptune import new as neptune
+        from neptune.new.run import Run
+        from neptune.new.exceptions import NeptuneLegacyProjectException
+    except ImportError:
+        import neptune
+        from neptune.run import Run
+        from neptune.exceptions import NeptuneLegacyProjectException
 else:
-    # needed for test mocks, these tests shall be updated
-    neptune, Experiment = None, None
+    # needed for test mocks, and function signatures
+    neptune, Run = None, None
+
+
+log = logging.getLogger(__name__)
+
+INTEGRATION_VERSION_KEY = "source_code/integrations/pytorch-lightning"
+
+# kwargs used in NeptuneLogger for legacy client (current NeptuneLegacyLogger)
+LEGACY_NEPTUNE_INIT_KWARGS = [
+    "project_name",
+    "offline_mode",
+    "experiment_name",
+    "experiment_id",
+    "params",
+    "properties",
+    "upload_source_files",
+    "abort_callback",
+    "logger",
+    "upload_stdout",
+    "upload_stderr",
+    "send_hardware_metrics",
+    "run_monitoring_thread",
+    "handle_uncaught_exceptions",
+    "git_info",
+    "hostname",
+    "notebook_id",
+    "notebook_path",
+]
+
+# kwargs used in legacy NeptuneLogger from neptune-pytorch-lightning package
+LEGACY_NEPTUNE_LOGGER_KWARGS = [
+    "base_namespace",
+    "close_after_fit",
+]
 
 
 class NeptuneLogger(LightningLoggerBase):
```

Comment:
Mind adding as code-owner for this file: @shnela

Reply:
Yes, I'll definitely add our team as code owners, thanks for suggestion!

Reply:
Yes, so you get notified to issues and maybe we can find a way to triage issues directly to your team too.

Reply:
We will create a netpune label, so it is easier for your team to navigate issues.